src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public class UserDaoImpl implements UserDao {
+    private static SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public User save(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can not save user to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return user;
+    }
+
+    @Override
+    public User get(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
This line is redundant since `sessionFactory` is already initialized at the class level. Consider removing this line and using the class-level `sessionFactory` variable.